### DISCUSSION TITLE
库自带的icon如何使用?

### DISCUSSION THREAD

**Discussion by sqk0232** [irihitech/Semi.Avalonia#515] [Q&A] [answered]
如题,Semi里icon的使用是要添加resource吗?还是另外的包?
该如何是要Semi里带有的icon?

**Comment by zdpcdt** [comment, marked as answer]
还没有正式发布，今天会发包。

Semi只提供StreamGeometry类型的字形数据，用法是
```axaml
<PathIcon Data="{StaticResource SemiIconStar}" />
```

**Reply by sqk0232** [reply]
好嘞
昨天想用一下icon,对照着示例去尝试,怎么都搞不出来...

**Reply by zdpcdt** [reply]
新版本已发布，可以试试了

**Reply by sqk0232** [reply]
可以使用了!!!
有个疑问就是,该怎么使用ViewModel去控制View里的icon呢?

**Reply by zdpcdt** [reply]
> 可以使用了!!! 有个疑问就是,该怎么使用ViewModel去控制View里的icon呢?

这个问题太笼统了，我只能回答使用绑定就能很简单地控制。

**Reply by sqk0232** [reply]
谢谢!
元旦快乐!

**Reply by sqk0232** [reply]
在ViewModel里使用string控制View里的icon
目前使用的ValueConverter去转换,不知道这个方式是不是绕了orz

**Reply by chunxinQ** [reply]
<PathIcon Data="{StaticResource SemiIconStar}" />  这个用法可以正常使用，但是参照demo中，使用  <ToggleSwitch
                    
                     Command="{Binding ToggleThemeCommand}"
                     OnContent="{StaticResource SemiIconMoon}"
                     OffContent="{StaticResource SemiIconSun}" />或者其他button中content时，图标显示为Avalonia.Media.StreamGeometry
![image](https://github.com/user-attachments/assets/834ae6c4-e3cc-43f7-ad0a-00ed322c0acb)

**Reply by zdpcdt** [reply]
> 这个用法可以正常使用，但是参照demo中，使用 <ToggleSwitch
> 
> ```
>                  Command="{Binding ToggleThemeCommand}"
>                  OnContent="{StaticResource SemiIconMoon}"
>                  OffContent="{StaticResource SemiIconSun}" />或者其他button中content时，图标显示为Avalonia.Media.StreamGeometry
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/149149250/405037307-834ae6c4-e3cc-43f7-ad0a-00ed322c0acb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc0MzY1NTYsIm5iZiI6MTczNzQzNjI1NiwicGF0aCI6Ii8xNDkxNDkyNTAvNDA1MDM3MzA3LTgzNGFlNmM0LWUzY2MtNDNmNy1hZDBhLTAwZWQzMjJjMGFjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMVQwNTEwNTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YzNkNTA5NzJjZmJmMjg4YjhiNzIyZjI0NDc1NmVkZWFiNWYwNjNhMzQxYmZkMjM0NDVlMWQxMmFjOTIwYTA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Yg_QZnjyIeK-m489iOfIGLkcOC-fnfYPA6tp1WbOzcc)

这是正常的，ToggleSwitch如果想显示图标需要在Content里使用PathIcon。
demo里控制明暗主题的Theme Key并没有在Semi.Avalonia包里，你可以参考这个写法：[ToggleSwitch.axaml](https://github.com/irihitech/Semi.Avalonia/blob/main/demo/Semi.Avalonia.Demo/Themes/ToggleSwitch.axaml)
